code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -17,29 +17,19 @@
 	window_name = "Automatic Station Cleaner v1.2"
 	pass_flags = PASSMOB
 	path_image_color = "#993299"
-
-	var/blood = 1
-	var/trash = 0
-	var/pests = 0
-
-	var/list/target_types
-	var/obj/effect/decal/cleanable/target
-	var/max_targets = 50 //Maximum number of targets a cleanbot can ignore.
-	var/oldloc = null
-	var/closest_dist
-	var/closest_loc
-	var/failed_steps
-	var/next_dest
-	var/next_dest_loc
+	var/datum/goap_agent/cleanbot/goap_ai
 
 /mob/living/simple_animal/bot/cleanbot/Initialize()
 	. = ..()
-	get_targets()
 	icon_state = "cleanbot[on]"
 
-	var/datum/job/janitor/J = new/datum/job/janitor
-	access_card.access += J.get_access()
+	var/datum/job/captain/C = new/datum/job/captain
```

Comment:
...?

Reply:
All access cleanbot

Reply:
Cleanbots get all access, sounds fine to me.

Reply:
That would be unwise. Cleanbots can be summoned via PDA, therefore allowing them to be used as airlock openers by getting them to path through restricted areas.

This is why bots (temporarily) download the access levels of the summoner such that they can enter anywhere the summoner can, without allowing the bot to retain the access during patrol mode.

Reply:
add bot posterns to airlocks, so they can just wander through them.

Reply:
Quite frankly, intentionally making a mess in a room to get an unwitting bot to let you in sounds hilarious and the exact definition of emergent gameplay. This stays.

Reply:
Agreed, and in my experience you can just ask the AI to let you in somewhere and it'd be quicker than summoning a cleanbot. If for some reason the AI wouldn't let you in, it's probably a high-security, high-traffic area where you'd get caught before the cleanbot arrives anyway.

Reply:
:-1:

Reply:
@kevinz000 go home security

Reply:
but you dont have to make a mess, goofball, you just pda the cleanbot and it lets you into the bridge

Reply:
m8 the bot summon makes them go to your tile, they'd have to already be in the bridge

Reply:
Example on Boxstation: Move a cleanbot one of main entrance doors, then walk to the door on the opposite side. Summon the cleanbot, and it will path a straight line to you, opening the airlocks for you on both sides.

I created the access downloading system instead of giving bots all-access for this exact reason.

Reply:
i could just remove cleanbot summoning

Reply:
https://github.com/tgstation/tgstation/pull/35947
just look at this model.why not to send both coords and access via NTnet?anyway, you will fuck with goap longer, then airlocs will take to merge. So just make it a bit later.